src/capi_frontend/capi.cpp

```diff
@@ -683,6 +709,141 @@ OVMS_Status* OVMS_Inference(OVMS_Server* serverPtr, OVMS_InferenceRequest* reque
+    std::shared_ptr<ovms::ModelInstance> modelInstance;
+    std::unique_ptr<ovms::Pipeline> pipelinePtr;
+    ovms::Server& server = *reinterpret_cast<ovms::Server*>(serverPtr);
+    auto status = getModelInstance(server, servableName, servableVersion, modelInstance, modelInstanceUnloadGuard);
+
+    if (status == StatusCode::MODEL_NAME_MISSING) {
+        SPDLOG_DEBUG("Requested model: {} does not exist. Searching for pipeline with that name...", servableName);
+        PipelineDefinition* pipelineDefinition = nullptr;
+        std::unique_ptr<PipelineDefinitionUnloadGuard> unloadGuard;
+        status = getPipelineDefinition(server, servableName, &pipelineDefinition, unloadGuard);
+        if (!status.ok() || !pipelineDefinition) {
+            return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::MODEL_NAME_MISSING));
+        }
+        *servableMetadata = reinterpret_cast<OVMS_ServableMetadata*>(new ovms::ServableMetadata(servableName, servableVersion, pipelineDefinition->getInputsInfo(), pipelineDefinition->getOutputsInfo()));
+        return nullptr;
+    }
+    if (!status.ok()) {
+        if (modelInstance) {
+            //    INCREMENT_IF_ENABLED(modelInstance->getMetricReporter().reqFailGrpcPredict);
+        }
+        SPDLOG_INFO("Getting modelInstance or pipeline failed. {}", status.string());
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    *servableMetadata = reinterpret_cast<OVMS_ServableMetadata*>(new ovms::ServableMetadata(servableName, servableVersion, modelInstance->getInputsInfo(), modelInstance->getOutputsInfo()));
+    return nullptr;
+}
+
+OVMS_Status* OVMS_ServableMetadataGetInputsCount(OVMS_ServableMetadata* servableMetadata, uint32_t* count) {
+    if (servableMetadata == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_METADATA));
+    }
+    if (count == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_NUMBER));
+    }
+    ovms::ServableMetadata* metadata = reinterpret_cast<ovms::ServableMetadata*>(servableMetadata);
+    *count = metadata->getInputsInfo().size();
+    return nullptr;
+}
+
+OVMS_Status* OVMS_ServableMetadataGetOutputsCount(OVMS_ServableMetadata* servableMetadata, uint32_t* count) {
+    if (servableMetadata == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_METADATA));
+    }
+    if (count == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_NUMBER));
+    }
+    ovms::ServableMetadata* metadata = reinterpret_cast<ovms::ServableMetadata*>(servableMetadata);
+    *count = metadata->getOutputsInfo().size();
+    return nullptr;
+}
+OVMS_Status* OVMS_ServableMetadataGetInput(OVMS_ServableMetadata* servableMetadata, uint32_t id, const char** name, OVMS_DataType* datatype, size_t* dimCount, int64_t** shapeMin, int64_t** shapeMax) {
```

Comment:
```suggestion
OVMS_Status* OVMS_ServableMetadataGetInput(OVMS_ServableMetadata* servableMetadata, uint32_t id, const char** name, OVMS_DataType* datatype, size_t* dimCount, int64_t** shapeMinArr, int64_t** shapeMaxArr) {
```
What about adding such suffix to indicate its an array?